csharp/src/Apache.Arrow.Adbc/Extensions/StandardSchemaExtensions.cs

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Apache.Arrow.Types;
+
+namespace Apache.Arrow.Adbc.Extensions
+{
+    public static class StandardSchemaExtensions
+    {
+
+        /// <summary>
+        /// Validates a data array that its column number and types match a given schema.
+        /// </summary>
+        /// <param name="schema">The schema to validate against.</param>
+        /// <param name="data">The data array to validate.</param>
+        /// <exception cref="ArgumentException">Throws an exception if the number of columns or type data types in the data array do not match the schema fields.</exception>
+        public static IReadOnlyList<IArrowArray> Validate(this Schema schema, IReadOnlyList<IArrowArray> data)
```

Comment:
This does a very superficial comparison of types. For instance, it doesn't check the precision and scale of decimals, length of a fixed-size list, the units of dates, times or intervals, whether or not a union is dense or sparse, the name or nullability of a field, etc. That seems mostly okay for an internal method being used privately against a relatively-known schema, but problematic for a public function that's being exported for anyone to use.

How many of the gaps in the above list are features (for this particular use case) rather than flaws? Should this be standard functionality in the core Arrow library? It seems to overlap with what the (internal) class ArrayDataTypeComparer does.

Reply:
I'd be happy to use the existing class, if it were exposed. I could also make this better an more rigorous, if you believe it might be used outside of its current context.